libnetdata/libnetdata.c

```diff
@@ -1453,3 +1453,40 @@ void recursive_config_double_dir_load(const char *user_path, const char *stock_p
     freez(udir);
     freez(sdir);
 }
+
+// Returns the number of bytes read from the file if file_size is not NULL.
+// The actual buffer has an extra byte set to zero (not included in the count).
+char *read_by_filename(char *filename, long *file_size)
+{
+    FILE *f = fopen(filename, "r");
+    if (!f)
+        return NULL;
+    if (fseek(f, 0, SEEK_END) < 0) {
+        fclose(f);
+        return NULL;
+    }
+    long size = ftell(f);
+    if (size <= 0 || fseek(f, 0, SEEK_END) < 0) {
+        fclose(f);
+        return NULL;
+    }
+    char *contents = callocz(size + 1, 1);
+    if (!contents) {
```

Comment:
`callocz()` never fails.

Reply:
So you like to live dangerously Mr Bond.